data/TaskMemoryDAO.js

```diff
@@ -6,12 +6,12 @@ const matchesCriteria = (task, criteria) => Object.keys(criteria).reduce((matchi
   return false;
 }, true);
 
-const validateNewTaskProps = (props) => (props.title && props.course);
+const validateNewTaskProps = (props) => (props && props.title && props.course);
```

Comment:
On this one, I feel like I should still leave the check for props. The reason being, if props is ever passed in as null I get a TypeError: Cannot read property 'title' of null. But I guess if null is ever passed in, the test cases won't pass due to the error. Would you leave this check in as a just in case? Or would you take out this test:
`chai.assert.isNotOk(dao.newTask(null));`
And leave it up to future developers to make sure they don't pass null into dao.newTask()?

Reply:
Yeah that's where you would need to make the decision I suppose. This is the safest way if you're type checking - though I would typically expect an error to be thrown if null was passed in. Personally I would leave it to the developer, and let the error happen if null was used.

That brings up the question though: If the properties are invalid do you want to just return out of the create and update, or should you provide an error? As it stands the code won't stop but it doesn't really provide feedback as to why the return value is undefined.